port/webhook/resource.go

```diff
@@ -0,0 +1,143 @@
+package webhook
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/port-labs/terraform-provider-port-labs/internal/cli"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &WebhookResource{}
+var _ resource.ResourceWithImportState = &WebhookResource{}
+
+func NewWebhookResource() resource.Resource {
+	return &WebhookResource{}
+}
+
+type WebhookResource struct {
+	portClient *cli.PortClient
+}
+
+func (r *WebhookResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
```

Comment:
i think this would be more appropriate (relevant to the whole file)
```suggestion
func (wr *WebhookResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
```